src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/framework/api/summaries/temperature/FullNodeTemperatureSummary.java

```diff
@@ -0,0 +1,131 @@
+
+    private final String nodeId;
+    private final String hostAddress;
+
+    public FullNodeTemperatureSummary(String nodeId, String hostAddress) {
+        this.nodeId = nodeId;
+        this.hostAddress = hostAddress;
+        this.nodeDimensionProfiles =
+                new DimensionalTemperatureSummary[TemperatureVector.Dimension.values().length];
+        this.temperatureVector = new TemperatureVector();
+    }
+
+    public TemperatureVector getTemperatureVector() {
+        return temperatureVector;
+    }
+
+    public List<DimensionalTemperatureSummary> getNodeDimensionProfiles() {
+        return Arrays.asList(nodeDimensionProfiles);
+    }
+
+    public String getNodeId() {
+        return nodeId;
+    }
+
+    public String getHostAddress() {
+        return hostAddress;
+    }
+
+    public void updateNodeDimensionProfile(DimensionalTemperatureSummary nodeDimensionProfile) {
+        TemperatureVector.Dimension dimension = nodeDimensionProfile.getProfileForDimension();
+        this.nodeDimensionProfiles[dimension.ordinal()] = nodeDimensionProfile;
+        temperatureVector.updateTemperatureForDimension(dimension, nodeDimensionProfile.getMeanTemperature());
+    }
+
+    public List<GenericSummary> getNestedSummaryList() {
+        List<GenericSummary> dimensionalSummaries = new ArrayList<>();
+        for (DimensionalTemperatureSummary dimSummary : nodeDimensionProfiles) {
+            dimensionalSummaries.add(dimSummary);
+        }
+        return dimensionalSummaries;
+    }
+
+    @Override
+    public <T extends GeneratedMessageV3> T buildSummaryMessage() {
+        throw new IllegalArgumentException();
+    }
+
+    @Override
+    public void buildSummaryMessageAndAddToFlowUnit(FlowUnitMessage.Builder messageBuilder) {
+        throw new IllegalArgumentException();
+    }
+
+    @Override
+    public String getTableName() {
+        return this.getClass().getSimpleName();
+    }
+
+    @Override
+    public List<Field<?>> getSqlSchema() {
+        List<Field<?>> schema = new ArrayList<>();
+        for (TemperatureVector.Dimension dimension: TemperatureVector.Dimension.values()) {
+            schema.add(DSL.field(DSL.name(dimension.NAME), Short.class));
+        }
+        return schema;
+    }
+
+    @Override
+    public List<Object> getSqlValue() {
+        List<Object> values = new ArrayList<>();
+        for (TemperatureVector.Dimension dimension: TemperatureVector.Dimension.values()) {
+            values.add(temperatureVector.getTemperatureFor(dimension));
+        }
+        return values;
+    }
+
+    @Override
+    public JsonElement toJson() {
+        JsonObject summaryObj = new JsonObject();
+        for (TemperatureVector.Dimension dimension: TemperatureVector.Dimension.values()) {
+            summaryObj.addProperty(dimension.NAME,
+                    temperatureVector.getTemperatureFor(dimension).getPOINTS());
+        }
+        getNestedSummaryList().forEach(
```

Comment:
this was a bug found and fixed in my previous PR. It needs to be updated as :

if (!this.nestedSummaryList.isEmpty()) {
      String tableName = this.nestedSummaryList.get(0).getTableName();
      summaryObj.add(tableName, this.nestedSummaryListToJson());
    }

Reply:
Thanks a lot @rguo-aws